libs/pilight/config/devices.c

```diff
@@ -255,9 +255,15 @@ int devices_update(char *protoname, JsonNode *json, enum origin_t origin, JsonNo
 									if(strcmp(sptr->name, opt->name) == 0
 									   && (opt->conftype == DEVICES_VALUE || opt->conftype == DEVICES_OPTIONAL)
 									   && opt->argtype == OPTION_HAS_VALUE) {
-										memset(vstring_, '\0', sizeof(vstring_));
 										vnumber_ = -1;
 										if(json_find_string(message, opt->name, &stmp) == 0) {
+ 											if(strlen(stmp) + 1 > vsize) {
```

Comment:
Can you increase the size in larger steps to prevent constant increasing of the size.

Reply:
I doubt if that is really necessary, because the initial size is 255. Only in rare cases strings will be longer than 254 and the chance of having more than one of such strings in one device is very small.

Reply:
Increasing the size by just 1 is almost always a bad idea.

Reply:
But then I'm afraid that you didn't really see what I did. 

I am not incrementing by 1. The new allocated size is always the length of the string that doesn't fit  (plus 1 for the terminator ofc).  The +1 you are pointing at is in the if statement, not in the realloc.

Reply:
You are right. Sorry. But, then increase by a fixed number above the length of the max string. E.g. steps of 255.

Reply:
Ofc I can do that, but imho then we are needlesly waisting memory, because the chance that another string of the same device is longer than the one that makes it necessary to enlarge the buffer, is almost zero.

For instance a label text can be very long, but the other strings (color and in my case also blink and bgcolor) are never that long and will always fit. And the same goes for the http protocol: the data returned can be very long, but the other strings (mimetype and state) never are. And I can't think of any feasable protocol that would have multiple strings longer than 254 chars.